pkg/parser/common.go

```diff
@@ -0,0 +1,309 @@
+			v.ReferTable.Name = model.NewCIStr(targetTableNames[1].Name)
+		}
+
+	case *ast.DropTableStmt:
+		if len(v.Tables) > 1 {
+			return "", ErrDropMultipleTables
+		}
+
+		v.Tables[0].Schema = model.NewCIStr(targetTableNames[0].Schema)
+		v.Tables[0].Name = model.NewCIStr(targetTableNames[0].Name)
+
+	case *ast.TruncateTableStmt:
+		v.Table.Schema = model.NewCIStr(targetTableNames[0].Schema)
+		v.Table.Name = model.NewCIStr(targetTableNames[0].Name)
+
+	case *ast.DropIndexStmt:
+		v.Table.Schema = model.NewCIStr(targetTableNames[0].Schema)
+		v.Table.Name = model.NewCIStr(targetTableNames[0].Name)
+	case *ast.CreateIndexStmt:
+		v.Table.Schema = model.NewCIStr(targetTableNames[0].Schema)
+		v.Table.Name = model.NewCIStr(targetTableNames[0].Name)
+	case *ast.RenameTableStmt:
+		if len(v.TableToTables) > 1 {
+			return "", ErrRenameMultipleTables
+		}
+
+		v.TableToTables[0].OldTable.Schema = model.NewCIStr(targetTableNames[0].Schema)
+		v.TableToTables[0].OldTable.Name = model.NewCIStr(targetTableNames[0].Name)
+		v.TableToTables[0].NewTable.Schema = model.NewCIStr(targetTableNames[1].Schema)
+		v.TableToTables[0].NewTable.Name = model.NewCIStr(targetTableNames[1].Name)
+
+	case *ast.AlterTableStmt:
+		if len(v.Specs) > 1 {
+			return "", ErrAlterMultipleTables
+		}
+
+		v.Table.Schema = model.NewCIStr(targetTableNames[0].Schema)
+		v.Table.Name = model.NewCIStr(targetTableNames[0].Name)
+
+		if v.Specs[0].Tp == ast.AlterTableRenameTable {
+			v.Specs[0].NewTable.Schema = model.NewCIStr(targetTableNames[1].Schema)
+			v.Specs[0].NewTable.Name = model.NewCIStr(targetTableNames[1].Name)
+		}
+
+	default:
+		return "", errors.Errorf("unkown type ddl %+v", stmt)
+	}
+
+	var b []byte
+	bf := bytes.NewBuffer(b)
+	err := stmt.Restore(&format.RestoreCtx{
+		Flags: format.DefaultRestoreFlags,
+		In:    bf,
+	})
+	if err != nil {
+		return "", errors.Annotate(err, "restore ast node")
+	}
+
+	return bf.String(), nil
+}
+
+// SplitDDL splits multiple operations in one DDL statement into multiple DDL statements
+// if fail to restore, it would not restore the value of `stmt` (it changes it's values if `stmt` is one of  DropTableStmt, RenameTableStmt, AlterTableStmt)
+func SplitDDL(stmt ast.StmtNode, schema string) (sqls []string, err error) {
+	var (
+		b          []byte
+		schemaName = model.NewCIStr(schema) // fill schema name
+		bf         = bytes.NewBuffer(b)
+		ctx        = &format.RestoreCtx{
+			Flags: format.DefaultRestoreFlags,
+			In:    bf,
+		}
+	)
+
+	switch v := stmt.(type) {
+	case *ast.CreateDatabaseStmt:
+		v.IfNotExists = true
+	case *ast.DropDatabaseStmt:
+		v.IfExists = true
+	case *ast.DropTableStmt:
+		v.IfExists = true
+
+		tables := v.Tables
+		for _, t := range tables {
+			if t.Schema.O == "" {
+				t.Schema = schemaName
+			}
+
+			v.Tables = []*ast.TableName{t}
+			bf.Reset()
+			err = stmt.Restore(ctx)
+			if err != nil {
+				return nil, errors.Annotate(err, "restore ast node")
```

Comment:
is it needed to `v.Tables = tables` before this `return`? similar work for other cases.

Reply:
according to comment in L179: if fail to restore, it would not restore the value of `stmt`.  but we can do it, let me see